assembly/src/assembler/basic_block_builder.rs

```diff
@@ -135,7 +135,7 @@ impl BasicBlockBuilder {
             let decorators = self.decorators.drain(..).collect();
 
             let basic_block_node = MastNode::new_basic_block_with_decorators(ops, decorators);
-            let basic_block_node_id = mast_forest_builder.ensure_node(basic_block_node);
+            let basic_block_node_id = mast_forest_builder.ensure_node(basic_block_node).unwrap();
```

Comment:
Shouldn't this `unwrap()` also be an error? It does complicate the return type a bit - but I'm not sure if there is a good way to get around this.

Reply:
Good catch, I missed this!